web/modules/custom/joinup_core/tests/src/ExistingSite/ShortIdAllowedValuesTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\joinup_core\ExistingSite;
+
+use Drupal\Core\Validation\Plugin\Validation\Constraint\RegexConstraint;
+use Drupal\Tests\joinup_test\ExistingSite\JoinupExistingSiteTestBase;
+use Drupal\rdf_entity\Entity\Rdf;
+
+/**
+ * Tests the short ID allowed values.
+ */
+class ShortIdAllowedValuesTest extends JoinupExistingSiteTestBase {
+
+  /**
+   * Tests that empty short ID is allowed.
+   */
+  public function testEmptyShortId(): void {
+    foreach (['collection', 'solution'] as $bundle) {
+      $entity = Rdf::create(['rid' => $bundle]);;
+      $violations = $entity->validate()->findByCodes([RegexConstraint::REGEX_FAILED_ERROR]);
+      $violations = iterator_to_array($violations);
+      $this->assertEmpty($violations);
+    }
+  }
+
+  /**
+   * Tests the short ID field and possible values.
+   *
+   * @dataProvider shortIdAllowedValuesProvider
+   */
+  public function testShortIdAllowedValues(string $short_id, bool $violation_expected): void {
+    foreach (['collection', 'solution'] as $bundle) {
+      $entity = Rdf::create([
+        'rid' => $bundle,
+        'field_short_id' => $short_id,
+      ]);;
+
+      $violations = $entity->validate()->findByCodes([RegexConstraint::REGEX_FAILED_ERROR]);
+      $violations = iterator_to_array($violations);
+
+      $this->assertCount((int) $violation_expected, $violations);
+    }
+  }
+
+  /**
+   * Data provider for ::testShortIdAllowedValues().
+   */
+  public function shortIdAllowedValuesProvider(): array {
+    return [
+      'starts with dash' => ['-StartWithDash', FALSE],
+      'ends with dash' => ['EndWithDash-', FALSE],
```

Comment:
Let's document that `FALSE` means that it is considered valid, and `TRUE` is considered invalid.